src/Test/Utilities/CoreClr/SharedConsoleOutWriter.cs

```diff
@@ -11,25 +11,35 @@ internal static class SharedConsole
         private static TextWriter s_savedConsoleOut;
         private static TextWriter s_savedConsoleError;
 
-        private static AsyncLocal<StringWriter> s_currentOut;
-        private static AsyncLocal<StringWriter> s_currentError;
+        private static AsyncLocal<TextWriter> s_currentOut;
+        private static AsyncLocal<TextWriter> s_currentError;
 
         internal static void OverrideConsole()
         {
             s_savedConsoleOut = Console.Out;
             s_savedConsoleError = Console.Error;
 
-            s_currentOut = new AsyncLocal<StringWriter>();
-            s_currentError = new AsyncLocal<StringWriter>();
+            s_currentOut = new AsyncLocal<TextWriter>();
+            s_currentError = new AsyncLocal<TextWriter>();
 
             Console.SetOut(new SharedConsoleOutWriter());
             Console.SetError(new SharedConsoleErrorWriter());
         }
 
-        public static void CaptureOutput(Action action, int expectedLength, out string output, out string errorOutput)
+        public static void CaptureOutput(Action action, int? expectedLength, out string output, out string errorOutput)
         {
-            var outputWriter = new CappedStringWriter(expectedLength);
-            var errorOutputWriter = new CappedStringWriter(expectedLength);
+            TextWriter errorOutputWriter;
+            TextWriter outputWriter;
+            if (expectedLength is int exLength)
+            {
+                errorOutputWriter = new CappedStringWriter(exLength);
+                outputWriter = new CappedStringWriter(exLength);
+            }
+            else
```

Comment:
Just because there's no expected output doesn't mean we shouldn't capture the incorrect output. I would keep always using CappedStringWriter.